README.md

```diff
@@ -628,6 +602,127 @@ const restServer = new MswServer({
 });
 ```
 
+## Concepts
+
+Behind a simple API (`getXXXHandler`), FakeRest uses a modular architecture that lets you combine different components to build a fake REST server that fits your needs.
+
+### Mocking Adapter
+
+`getXXXHandler` is a shortcut to an object-oriented API:
+
+```js
+export const getMswHandler = (options: MswAdapterOptions) => {
+    const server = new MswAdapter(options);
+    return server.getHandler();
+};
+```
+
+FakeRest provides 3 adapter classes:
+
+- `MswAdapter`: Based on [MSW](https://mswjs.io/)
+- `FetchMockAdapter`: Based on [`fetch-mock`](https://www.wheresrhys.co.uk/fetch-mock/)
+- `SinonAdapter`: Based on [Sinon](https://sinonjs.org/releases/v18/fake-xhr-and-server/)
+
+You can use the adapter class directly, e.g. if you want to store the adapter instance in a variable:
+
+```js
+import { MsWAdapter } from 'fakerest';
+
+const adapter = new MswAdapter({
+    baseUrl: 'http://my.custom.domain',
+    data: {
+        'authors': [
+            { id: 0, first_name: 'Leo', last_name: 'Tolstoi' },
+            { id: 1, first_name: 'Jane', last_name: 'Austen' }
+        ],
+        'books': [
+            { id: 0, author_id: 0, title: 'Anna Karenina' },
+            { id: 1, author_id: 0, title: 'War and Peace' },
+            { id: 2, author_id: 1, title: 'Pride and Prejudice' },
+            { id: 3, author_id: 1, title: 'Sense and Sensibility' }
+        ],
+        'settings': {
+            language: 'english',
+            preferred_format: 'hardback',
+        }
+    }
+});
+window.fakerest = adapter;
+const handler = adapter.getHandler();
+```
+
+### REST Server
+
+The REST syntax is implemented by a Server object. It takes a normalized request and returns a normalized response. Adapters rely on the server to handle the REST API.
+
+FakeRest currently provides only one server implementation: `SimpleRestServer`.
+
+You can specify the server to use in an adapter by passing the `server` option:
+
+```js
+const server = new SimpleRestServer({
+    baseUrl: 'http://my.custom.domain',
+    data: {
+        'authors': [
+            { id: 0, first_name: 'Leo', last_name: 'Tolstoi' },
+            { id: 1, first_name: 'Jane', last_name: 'Austen' }
+        ],
+        'books': [
+            { id: 0, author_id: 0, title: 'Anna Karenina' },
+            { id: 1, author_id: 0, title: 'War and Peace' },
+            { id: 2, author_id: 1, title: 'Pride and Prejudice' },
+            { id: 3, author_id: 1, title: 'Sense and Sensibility' }
+        ],
+        'settings': {
+            language: 'english',
+            preferred_format: 'hardback',
+        }
+    }
+});
+const adapter = new MswAdapter({ server });
+const handler = adapter.getHandler();
+```
+
```

Comment:
IMO we should go a bit further and at least mention the required methods of a `Server` object and their input/output